Add pg_dist_authinfo schema and validation

jasonmp85 opened the pull request:
This table will be used by Citus Enterprise to populate authentication-related fields in outbound connections; Citus Community lacks support for this functionality.

marcocitus commented on src/backend/distributed/utils/metadata_cache.c:
```diff
@@ -3415,3 +3419,26 @@ DistNodeMetadata(void)
+	ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+					errmsg("catalog table unsupported"),
+					errdetail("Citus Community Edition does not support the use of per-"
+							  "host/user authentication options."),
```
`per host/` is probably not relevant here

jasonmp85 replied:
Do we want to maybe let Community users at least specify a password here? Using `pgpass` is still kind of annoying and requires filesystem access. I think I wrote this error message with the idea that we might let them set at least a global password.

marcocitus commented on src/backend/distributed/utils/metadata_cache.c:
```diff
@@ -3415,3 +3419,26 @@ DistNodeMetadata(void)
+authinfo_valid(PG_FUNCTION_ARGS)
+{
+	ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+					errmsg("catalog table unsupported"),
```
Maybe `cannot write to pg_dist_authinfo`. That's a little clearer in case this call happens somewhere deep down.

jasonmp85 replied:
Sure.

jasonmp85 commented:
I've addressed all your comments and added a test to exercise the check constraint implementations. We good to :shipit: (i.e. merge this into the other PR?)

jasonmp85 commented:
@marcocitus I don't know what's up with appveyor, but I've addressed any remaining feedback and documented every method; is this good to go, or is anything else needed? I responded to your questions about poolname/poolport in our email thread.

jasonmp85 commented:
@marcocitus Also, Appveyor is getting a weird error here for `dml_recursive`; it did it in both branches. I don't immediately see why this would be… could there be something non-deterministic about that test?

jasonmp85 commented:
I re-triggered the branch build here (but not on #2190) so I suppose we'll see how non-deterministic it is.
